core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1354,6 +1395,22 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
 
   }
 
+  case class LogDirEventNotification(sequenceNumbers: Seq[String]) extends ControllerEvent {
+
+    def state = ControllerState.LogDirChange
+
+    override def process(): Unit = {
```

Comment:
In controller failover, would it be worth to clean up all leftover sequence nodes in logDirEventPath? Those nodes won't be useful since the new controller will send a LeaderAndIsrRequest to every broker on failover.